routes/api.js

```diff
@@ -1 +1,85 @@
-var express=require("express"),router=express.Router(),mongoose=require("mongoose"),Recipe=mongoose.model("Recipe");router.get("/recipes",function(e,o){Recipe.find(function(e,d){e&&o.send(500,e),o.send(d)})}),router.get("/recipe/:id",function(e,o){Recipe.findById(e.params.id,function(e,d){e&&o.send(500,e),o.json(d)}),console.log(e.params.id)}),router.post("/recipes",function(e,o){var d=new Recipe;d.title=e.body.title,d.descript=e.body.descript,d.ingredients=e.body.ingredients,d.body=e.body.body,d.type=e.body.type,d.posted_by=e.body.posted_by,d.save(function(e,d){return e&&o.send(500,e),o.json(d)})}),router.put("/recipe/:id",function(e,o){Recipe.findById(e.params.id,function(d,i){d&&o.send(500,d),i.title=e.body.title,i.descript=e.body.descript,i.ingredients=e.body.ingredients,i.body=e.body.body,i.type=e.body.type,i.posted_by=e.body.posted_by,i.save(function(e,d){return e&&o.send(500,e),o.json(d)})})}),router.delete("/recipe/:id",function(e,o){Recipe.remove({_id:e.params.id},function(e,d){e&&o.send(500,e),o.send(d)})}),module.exports=router;
\ No newline at end of file
+var express = require("express");
+var router = express.Router();
+var mongoose = require("mongoose");
+var Recipe = mongoose.model("Recipe");
+
+router.get("/recipes", function(req, res) {
+    Recipe.find(function(err, data) {
+        if (err) {
+            res.send(500, err);
+        }
+        res.send(data);
+    });
+});
+
+router.get("/recipe/:id", function(req, res) {
+    Recipe.findById(req.params.id, function(err, recipe) {
+        if (err) {
+            res.send(500, err);
+        }
+        res.json(recipe);
```

Comment:
It's dangerous to just retrieve whole object from database, more correct to use DTO approach, which has some specifics in NodeJS case: https://stackoverflow.com/a/31227246